emission/analysis/intake/segmentation/trip_segmentation_methods/dwell_segmentation_dist_filter.py

```diff
@@ -46,102 +50,201 @@ def segment_into_trips(self, timeseries, time_query):
         data that they want from the sensor streams in order to determine the
         segmentation points.
         """
-        self.filtered_points_df = timeseries.get_data_df("background/filtered_location", time_query)
-        self.filtered_points_df.loc[:,"valid"] = True
-        self.transition_df = timeseries.get_data_df("statemachine/transition", time_query)
+        with ect.Timer() as t_get_filtered_points:
+            self.filtered_points_df = timeseries.get_data_df("background/filtered_location", time_query)
+            user_id = self.filtered_points_df["user_id"].iloc[0]
+        esds.store_pipeline_time(
+            user_id,
+            ecwp.PipelineStages.TRIP_SEGMENTATION.name + "/segment_into_trips/get_filtered_points_df",
+            time.time(),
+            t_get_filtered_points.elapsed
+        )
+
+        with ect.Timer() as t_mark_valid:
+            self.filtered_points_df.loc[:, "valid"] = True
+        esds.store_pipeline_time(
+            user_id,
+            ecwp.PipelineStages.TRIP_SEGMENTATION.name + "/segment_into_trips/mark_valid",
+            time.time(),
+            t_mark_valid.elapsed
+        )
+
+        with ect.Timer() as t_get_transition_df:
+            self.transition_df = timeseries.get_data_df("statemachine/transition", time_query)
+        esds.store_pipeline_time(
+            user_id,
+            ecwp.PipelineStages.TRIP_SEGMENTATION.name + "/segment_into_trips/get_transition_df",
+            time.time(),
+            t_get_transition_df.elapsed
+        )
+
         if len(self.transition_df) > 0:
             logging.debug("self.transition_df = %s" % self.transition_df[["fmt_time", "transition"]])
         else:
             logging.debug("no transitions found. This can happen for continuous sensing")
 
-        self.last_ts_processed = None
+        with ect.Timer() as t_initialize:
+            self.last_ts_processed = None
+        esds.store_pipeline_time(
+            user_id,
+            ecwp.PipelineStages.TRIP_SEGMENTATION.name + "/segment_into_trips/initialize_last_ts_processed",
+            time.time(),
+            t_initialize.elapsed
+        )
```

Comment:
This is overkill. Why would this need to be its own step?
I don't think there's any situation where assigning a primitive value is going to take a significant amount of time

Reply:
I had also linked to several others earlier (single log, creating a dictionary)